### PR TITLE
GRW-1838 / Feature / Use Money object for price data and formatting

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Always take currency from money object instead of some context

## Justify why they are needed

Explicit is better than implicit.  Use same data shape or our GraphQL API - less transform code, lower mental load

## Checklist before requesting a review

- [x] I have performed a self-review of my code

### CODE REVIEW

File: apps/store/src/components/CartInventory/CartInventory.types.ts [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
   title: string
-  cost: number
-  currencyCode: string
+  cost: Money
   startDate?: Date
 }
 
 export type CartCampaign = { id: string; displayName: string }
 
 export type CartCost = {
```
Comment (alebedev) [inline apps/store/src/components/CartInventory/CartInventory.types.ts:12]: Maybe we should use CartCost from the API?
```
{ net: Money, gross: Money, discount: Money }
```

Comment (robinandeer) [reply apps/store/src/components/CartInventory/CartInventory.types.ts:12]: Hmm, I prefer to transform data and send it to dumb components 🙃

File: apps/store/src/utils/formatter.ts [hunk 2]
```diff
@@ -2,22 +2,27 @@ import { i18n as I18NextClient } from 'i18next'
 import { CurrencyCode } from '@/services/apollo/generated'
 import { IsoLocale } from '@/utils/l10n/types'
 
-type MoneyFormatOptions = { currencyCode: CurrencyCode; locale: IsoLocale }
+type MoneyFormatOptions = { locale: IsoLocale }
 
-export const formatAmount = (amount: number, options: MoneyFormatOptions): string => {
+export type Money = {
```
Comment (robinandeer) [inline apps/store/src/utils/formatter.ts:7]: nit: alias the type generated from Apollo?

**Review by robinandeer — APPROVED** [review]
I Like It

File: apps/store/src/utils/formatter.ts (hunk 2 above)
Comment (alebedev) [reply apps/store/src/utils/formatter.ts:7]: The problem with alias is that you now have 2 places to import from - IDEs may pick random one

**Pull request merged**